### PR TITLE
Dreamchecker Spawn return fix

### PR DESCRIPTION
**Pull request by Hughgent**
## About The Pull Request

Dreamchecker is complaining about these five things returning values because they are behind a `spawn()`

image

## Why It's Good For The Game

No change to the overall game

## Changelog
:cl: Hughgent
code: removes the warnings of the codebase related to spawn()
/:cl:

### CODE REVIEW

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Fix the spawn()'s instead please this is a bandaid

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -364,14 +364,16 @@
...
+			addtimer(CALLBACK(src, .proc/throw_impact_async), 2)
 			if(isliving(src))
 				var/mob/living/M = src
 				M.turf_collision(T, speed)
 
 	SEND_SIGNAL(src, COMSIG_MOVABLE_IMPACT, hit_atom)
 
+///handles a step of a thing hitting a wall like other thing, the bounce back.
+/atom/movable/proc/throw_impact_async()
+	step(src, turn(dir, 180))
```
Comment (TiviPlus) [inline code/game/atoms_movable.dm:376]: Why not call step direct?

File: code/game/objects/effects/misc.dm [hunk 2]
```diff
@@ -57,10 +57,7 @@
 		setDir(ndir)
 		..(nloc)
 
-		spawn(20)
-			loc = null
-
-
+		addtimer(VARSET_CALLBACK(src, loc, null), 2 SECONDS)
```
Comment (TiviPlus) [inline code/game/objects/effects/misc.dm:60]: Just delete this effect tbh is pretty shit and goes unused

File: code/game/objects/items/handcuffs.dm [hunk 3]
```diff
@@ -135,7 +135,9 @@
...
 				return
 
-		spawn(30)
-			if(!C)	return
-			if(p_loc == user.loc && p_loc_m == C.loc)
-				C.update_handcuffed(new /obj/item/restraints/handcuffs(C))
+		addtimer(CALLBACK(src, .proc/make_handcuffed, C, user, p_loc, p_loc_m), 3 SECONDS)
+
+/obj/item/restraints/handcuffs/cyborg/proc/make_handcuffed(mob/living/carbon/C as mob, mob/user as mob, p_loc, p_loc_m)
+	if(!C)	return
```
Comment (TiviPlus) [inline code/game/objects/items/handcuffs.dm:141]: ahem

File: code/game/objects/items/handcuffs.dm [hunk 4]
```diff
@@ -135,7 +135,9 @@
 				to_chat(user, "<span class='warning'>\The [H] needs at least two wrists before you can cuff them together!</span>")
 				return
 
-		spawn(30)
-			if(!C)	return
-			if(p_loc == user.loc && p_loc_m == C.loc)
-				C.update_handcuffed(new /obj/item/restraints/handcuffs(C))
+		addtimer(CALLBACK(src, .proc/make_handcuffed, C, user, p_loc, p_loc_m), 3 SECONDS)
+
+/obj/item/restraints/handcuffs/cyborg/proc/make_handcuffed(mob/living/carbon/C as mob, mob/user as mob, p_loc, p_loc_m)
```
Comment (TiviPlus) [inline code/game/objects/items/handcuffs.dm:140]: Dont use the as mob things

File: code/modules/flufftext/Dreaming.dm [hunk 5]
```diff
@@ -17,15 +17,18 @@
...
+///handles the display of dream text to the player.
+mob/living/carbon/proc/dream_async()
+	for(var/i = rand(1,4),i > 0, i--)
+		to_chat(src, "<span class='notice'><i>... [pick(POSSIBLE_DREAM_TOPICS)] ...</i></span>")
+		sleep(rand(40,70))
+		if(!IsUnconscious())
+			dreaming = 0
+			return
+	dreaming = 0
+	return
```
Comment (TiviPlus) [inline code/modules/flufftext/Dreaming.dm:31]: 4 things: 
forgot the / in front of mob
this for loop isnt defined right you should use our stand
Dreaming is a bool
This sleep causes hard deletes ping me if you want more explained, but it needs to  be another proc

File: code/game/atoms_movable.dm (hunk 1 above)
Comment (Hughgent) [reply code/game/atoms_movable.dm:376]: to simulate the 'bounce' I presume.

Comment (TiviPlus) [reply code/game/atoms_movable.dm:376]: That's not how code works

**Pull request closed without merging**